JamGame/Assets/Scripts/Level/Boss/Task.cs

```diff
@@ -4,14 +4,21 @@
 
 namespace Level.Boss.Task
 {
+    public struct Progress
+    {
+        public float Completion;
+        public float Overall;
+        public bool Complete;
+    }
+
     [InterfaceEditor]
     public interface ITask
     {
         public void ValidateProviders();
 
         public void Update(float delta_time) { }
 
-        public bool IsComplete();
+        public Progress GetProgress();
```

Comment:
Why you don't want do this as a getter and setter property?
For put away returning new instance every time like in line 110